### PR TITLE
chore(ui-grid): add TS types to `ui-grid` package

### PR DESCRIPTION
**Pull request by ke1k0**
Closes: INSTUI-3053

### CODE REVIEW

File: packages/ui-grid/src/GridCol/styles.ts [hunk 1]
```diff
@@ -111,31 +113,29 @@ const generateStyle = (
...
-  const getColSize = (breakpoint: GridBreakpoints) => {
+  const getColSize = (breakpoint: BreakPoints) => {
     let { width } = props
 
     if (!width) return
 
     if (width && typeof width === 'object') {
-      // @ts-expect-error TODO: type when typing component
-      width = width[breakpoint]
+      width = width[breakpoint === 'x-large' ? 'xLarge' : breakpoint]
```
Comment (matyasf) [inline packages/ui-grid/src/GridCol/styles.ts:126]: was this a bug that you found while doing the types?

File: packages/ui-grid/src/GridRow/props.ts [hunk 2]
```diff
@@ -38,6 +38,9 @@ import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
 import type { GridBreakpoints } from '../GridTypes'
 
 type GridRowOwnProps = {
+  /**
+   * On of: `Grid.Col`, `ScreenReaderContent`
```
Comment (matyasf) [inline packages/ui-grid/src/GridRow/props.ts:42]: typo: On of

**Review by matyasf — APPROVED** [review]

File: packages/ui-grid/src/GridCol/styles.ts (hunk 1 above)
Comment (ke1k0) [reply packages/ui-grid/src/GridCol/styles.ts:126]: Yeah, it seems so, but there were no complaints about it before.

Comment (ke1k0) [reply packages/ui-grid/src/GridCol/styles.ts:126]: I didn't think it was important to mention as a "fix" commit and show it in the Changelog, do you think I should?

File: packages/ui-grid/src/GridRow/props.ts (hunk 2 above)
Comment (ke1k0) [reply packages/ui-grid/src/GridRow/props.ts]: Ooops, thanks, corrected!

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**